velox/connectors/hive/storage_adapters/gcs/GcsFileSystem.cpp

```diff
@@ -447,7 +447,34 @@ void GcsFileSystem::mkdir(
 }
 
 void GcsFileSystem::rmdir(std::string_view path) {
-  VELOX_UNSUPPORTED("rmdir for GCS not implemented");
+  if (!isGcsFile(path))
+    VELOX_FAIL(kGcsInvalidPath, path);
+
+  const auto file = gcsPath(path);
+  std::string bucket;
+  std::string object;
+  setBucketAndKeyFromGcsPath(file, bucket, object);
+  for (auto&& metadata : impl_->getClient()->ListObjects(bucket)) {
+    if (!metadata.ok()) {
```

Comment:
remove if since `checkGcsStatus` checks the status?

Reply:
@majetideepak Yes. Removed.